pkg/pillar/hypervisor/xen.go

```diff
@@ -0,0 +1,286 @@
+		return 0, fmt.Errorf("xl domid failed: %s\n",
+			string(stdoutStderr))
+	}
+	res := strings.TrimSpace(string(stdoutStderr))
+	domainID, err := strconv.Atoi(res)
+	if err != nil {
+		log.Errorf("Can't extract domainID from %s: %s\n", res, err)
+		return 0, fmt.Errorf("Can't extract domainID from %s: %s\n", res, err)
+	}
+	return domainID, nil
+}
+
+func (ctx xenContext) Start(domainName string, domainID int) error {
+	log.Infof("xlUnpause %s %d\n", domainName, domainID)
+	cmd := "xl"
+	args := []string{
+		"unpause",
+		domainName,
+	}
+	stdoutStderr, err := wrap.Command(cmd, args...).CombinedOutput()
+	if err != nil {
+		log.Errorln("xl unpause failed ", err)
+		log.Errorln("xl unpause output ", string(stdoutStderr))
+		return fmt.Errorf("xl unpause failed: %s\n",
+			string(stdoutStderr))
+	}
+	log.Infof("xlUnpause done. Result %s\n", string(stdoutStderr))
+	return nil
+}
+
+func (ctx xenContext) Stop(domainName string, domainID int, force bool) error {
+	log.Infof("xlShutdown %s %d\n", domainName, domainID)
+	cmd := "xl"
+	var args []string
+	if force {
+		args = []string{
+			"shutdown",
+			"-F",
+			domainName,
+		}
+	} else {
+		args = []string{
+			"shutdown",
+			domainName,
+		}
+	}
+	stdoutStderr, err := wrap.Command(cmd, args...).CombinedOutput()
+	if err != nil {
+		log.Errorln("xl shutdown failed ", err)
+		log.Errorln("xl shutdown output ", string(stdoutStderr))
+		return fmt.Errorf("xl shutdown failed: %s\n",
+			string(stdoutStderr))
+	}
+	log.Infof("xl shutdown done\n")
+	return nil
+}
+
+func (ctx xenContext) Delete(domainName string, domainID int) error {
+	log.Infof("xlDestroy %s %d\n", domainName, domainID)
+	cmd := "xl"
+	args := []string{
+		"destroy",
+		domainName,
+	}
+	stdoutStderr, err := wrap.Command(cmd, args...).CombinedOutput()
+	if err != nil {
+		log.Errorln("xl destroy failed ", err)
+		log.Errorln("xl destroy output ", string(stdoutStderr))
+		return fmt.Errorf("xl destroy failed: %s\n",
+			string(stdoutStderr))
+	}
+	log.Infof("xl destroy done\n")
+	return nil
+}
+
+func (ctx xenContext) Info(domainName string, domainID int) error {
+	log.Infof("xlStatus %s %d\n", domainName, domainID)
+	// XXX xl list -l domainName returns json. XXX but state not included!
+	// Note that state is not very useful anyhow
+	cmd := "xl"
+	args := []string{
+		"list",
+		"-l",
+		domainName,
+	}
+	stdoutStderr, err := wrap.Command(cmd, args...).CombinedOutput()
+	if err != nil {
+		log.Errorln("xl list failed ", err)
+		log.Errorln("xl list output ", string(stdoutStderr))
+		return fmt.Errorf("xl list failed: %s\n",
+			string(stdoutStderr))
+	}
+	// XXX parse json to look at state? Not currently included
+	// XXX note that there is a warning at the top of the combined
+	// output. If we want to parse the json we need to get Output()
+	log.Infof("xl list done. Result %s\n", string(stdoutStderr))
+	return nil
+}
+
+func (ctx xenContext) LookupByName(domainName string, domainID int) (int, error) {
```

Comment:
From an API perspective it makes sense to remove the domainID and instead have the caller do the "domainid changed" check.

Reply:
All true, but like I said -- I really didn't want to change a single sig to make the reviews and step #1 refactoring easier. We'll address this in follow up PRs.